src/smk/tool/geomark/tool-geomark.js

```
@@ -44,6 +67,31 @@ include.module( 'tool-geomark', [
 
                 geomarkLayers.addLayer(layer);
             });
+
+            var baseUrl = 'https://apps.gov.bc.ca/pub/geomark';
+            var client = new window.GeomarkClient(baseUrl);
```

I think the baseUrl value should be inlined to line 72 ... I forgot to do this before pushing.

---

Could we read the baseUrl from configuration instead? I'm not crazy about hard coded URLs whenever possible. It would be a little more 'future proof' in the event that the URL were to ever change for some reason.

---

I've been looking for examples of URLs in configuration, but it looks like hard-coding is the norm in this project. Do you know if we even have (non-build-related) configuration files?

---

I think the URL could be read from geomark config in smk-config.json. A hard coded default would then exist in the smk-cli project which would be written to the smk-config.json when adding the tool to an app.

---

I've added support for geomark config in smk-config.json. Here is a snippet:

            "geomarkService": {
                "url": "https://apps.gov.bc.ca/pub/geomark"
            }

The full tool config would now look like:

        {
            "title": "Geomark",
            "position": "list-menu",
            "icon": "build",
            "type": "geomark",
            "geomarkService": {
                "url": "https://apps.gov.bc.ca/pub/geomark"
            },
            "enabled": true
        },

I've made this an object instead of a single property after seeing this done elsewhere in SMK - for example, the Directions/Waypoints tool declares objects for "geocoderService" and "routePlannerService".

I've also added a check for its presence, and notification if it's missing, so we can fail more gracefully if it's missing in config.

---

Awesome, thank you!